lib/web/scripts/node-join/install.sh

```diff
@@ -758,87 +760,176 @@ if teleport_binaries_exist; then
-        ${PACKAGE_MANAGER_COMMAND} "${TEMP_DIR}/${DOWNLOAD_FILENAME}"
-    # use rpm if we couldn't find a package manager
+        log_important "Can't figure out what Teleport format to use"
+        exit 1
+    fi
+}
+
+install_from_repo() {
+    if [[ "${REPO_CHANNEL}" == "" ]]; then
+        # By default, use the current version's channel.
+        REPO_CHANNEL=v"${TELEPORT_VERSION//.*/}"
+    fi
+
+    # Populate $ID, $VERSION_ID, $VERSION_CODENAME and other env vars identifying the OS.
+    # shellcheck disable=SC1091
+    . /etc/os-release
+
+    if [ "$ID" == "debian" ] || [ "$ID" == "ubuntu" ]; then
+        # old versions of ubuntu require that keys get added by `apt-key add`, without
+        # adding the key apt shows a key signing error when installing teleport.
+        if [[
+            ($ID == "ubuntu" && $VERSION_ID == "16.04") || \
+            ($ID == "debian" && $VERSION_ID == "9" )
+        ]]; then
+            apt install apt-transport-https gnupg -y
+            curl -fsSL https://deb.releases.teleport.dev/teleport-pubkey.asc | apt-key add -
+            echo "deb https://apt.releases.teleport.dev/${ID} ${VERSION_CODENAME} stable/${REPO_CHANNEL}" > /etc/apt/sources.list.d/teleport.list
+        else
+            curl -fsSL https://deb.releases.teleport.dev/teleport-pubkey.asc \
+                -o /usr/share/keyrings/teleport-archive-keyring.asc
+            echo "deb [signed-by=/usr/share/keyrings/teleport-archive-keyring.asc] \
+            https://apt.releases.teleport.dev/${ID} ${VERSION_CODENAME} stable/${REPO_CHANNEL}" > /etc/apt/sources.list.d/teleport.list
+        fi
+        apt-get update
+        apt-get install -y ${TELEPORT_PACKAGE_NAME}
+    elif [ "$ID" = "amzn" ] || [ "$ID" = "rhel" ] || [ "$ID" = "centos" ] ; then
+        if [ "$ID" = "rhel" ]; then
+            VERSION_ID="${VERSION_ID//.*/}" # convert version numbers like '7.2' to only include the major version
+        fi
+        yum install -y yum-utils
+        yum-config-manager --add-repo \
+        "$(rpm --eval "https://yum.releases.teleport.dev/$ID/$VERSION_ID/Teleport/%{_arch}/stable/${REPO_CHANNEL}/teleport.repo")"
+        yum install -y ${TELEPORT_PACKAGE_NAME}
     else
-        # install RPM (in upgrade mode)
-        log "Using rpm to install ${TEMP_DIR}/${DOWNLOAD_FILENAME}"
-        rpm -Uvh "${TEMP_DIR}/${DOWNLOAD_FILENAME}"
+        echo "Unsupported distro: $ID"
+        exit 1
+    fi
+}
+
+is_repo_available() {
+    if [[ "${OSTYPE}" != "linux-gnu" ]]; then
+        return 1
+    fi
+
+    # Populate $ID, $VERSION_ID and other env vars identifying the OS.
+    # shellcheck disable=SC1091
+    . /etc/os-release
+
+    # Only the following distros are supported:
+    if [[ ! "$ID" =~ ^(ubuntu|debian|amzn|rhel|centos)$ ]]; then
+        return 1
+    fi
+
+    # Teleport's deb repo has the following Ubuntu Versions:
+    if [[ "$ID" == "ubuntu" && ! "$VERSION_ID" =~ ^(16\.04|18\.04|20\.04|22\.04) ]]; then
```

Comment:
Rather than all the `if` checks with multiple conditions, you could write this function like this:
```
is_repo_available() {
    if [[ "${OSTYPE}" != "linux-gnu" ]]; then
        return 1
    fi

    # Populate $ID, $VERSION_ID and other env vars identifying the OS.
    # shellcheck disable=SC1091
    . /etc/os-release

    # The following distros+version have a Teleport repository to install from.
    case "${ID}-${VERSION_ID}" in
        ubuntu-16.04* | ubuntu-18.04* | ubuntu-20.04* | ubuntu-22.04*)
            return 0 ;;
        debian-9* | debian-10* | debian-11*)
            return 0 ;;
        rhel-7* | rhel-8* | rhel-9*)
            return 0 ;;
        centos-7* | centos-8* | centos-9*)
            return 0 ;;
        amzn-2)
            return 0 ;;
    esac

    return 1 
}
```

It reads a bit clearer exactly which distros/versions support repo installation.

Note: I just wrote this here and didn't test it - just presenting it as an idea. Up to you if you prefer it or not.

Reply:
Sounds good :+1: 
Thank you